### DISCUSSION TITLE
Download Helper Windows Signature Certificate expired

### DISCUSSION THREAD

**Discussion by Wixlogo** [aclap-dev/video-downloadhelper#1088] [Help and Q&A]
### Your question

This is a reminder that the signature certificate for the installer of the downloader has expired.
The program has consistently been flagged as a false positive by AVs; IDK why. However, after the signature was expired, the AVs are going crazy with it!
![image](https://github.com/user-attachments/assets/1dbb9c31-2632-43cd-9793-880fa8816f72)

### Addon & CoApp version and details

I don't think this is required for this post.......

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3958, Firefox 128.0.3 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.19.

The date of 2024/2/19 is the date on which CoApp version 2.0.19 (a coincidence that there's so many 19s there) was released.  As I interpret it, the signature is applied to the installer program on the day it is released.  The signature is then carried in the executable for as long as you have it on your system.  @paulrouget Paul isn't going to push out a new signature every day to every user who has the installer program.  I believe the signature on the installer, which is what you are showing in your image, is not meant to change over time.  The signature just proves it's the real program.  The program gets signed once & that's supposed to satisfy anybody who downloads it whenever they download it.  At least, that's my view of what that signature is for.  It's not at all like certificates that web sites use to validate their web pages.

I don't think that signature is what is triggering the antivirus community.  The antivirus community is a separate issue having nothing to do with this signature.  There's been lots of discussion of antivirus on here.  You need to hunt that down & read a good sample of it.  The only antivirus I've ever used on my W11 box has been the Windows Defender built into the system.  When I say "ever" I mean since early last December, not all that long ago.  But that was when I finally got this machine configured & put the finishing touches on my successful install of W11.  I installed this version of the CoApp on this past February 19, if the date stamp on the installer file is any indication, in addition to the date stamps on the files in my CoApp directory.  It's been running fine ever since, & Windows Defender hasn't let out the slightest peep for the CoApp.  Other AV software might flag it, but this is a routine issue.  You're supposed to tell your AV to stop flagging the CoApp executables because YOU trust the CoApp.  Why do you trust the CoApp?  Because you got it from the authorized source, plus the installer has a proper signature.  AV software can't know about every piece of software everywhere in the world.  That's why every AV program provides a way to define exceptions so it stops interfering with the correct execution of programs you know to be trustworthy.  "The AVs are going crazy with it" is a situation that smart users calmly ignore.

**Reply by Wixlogo** [reply]
The file certificate/signature in Windows actually expires after a certain period I believe. Developers are supposed to renew them after they expire. For CoApp, the file signature was valid from 14-05-2021 to 14-05-2024. You can see this in the file details (it's August now, by the way).

This is how it is supposed to look (example from another app called BCUninstaller): when opening the details tab of the installer, you can see that the program will be signed till 10-05-2025. After that, it will show signature expired, which is exactly what happened with CoApp.
<img width="959" alt="image" src="https://github.com/user-attachments/assets/d3c532c5-f14e-4b21-89d6-7c33d6060f66">

I absolutely agree with your point about antivirus programs making false positives. It's the antivirus software showing incorrect results. CoApp is 100% safe and is open-source, so there is no doubt about its safety. My post wasn't really about antivirus false positives. The reason for the post was to remind that it's August 2024 and the previous signature has expired.

**Reply by GrampaWildWilly** [reply]
Well now you've got me intrigued.  Looking at the certificate on the latest Firefox installer (128.0.3) it says it's good until 2027.  The one on the CoApp has an initiation date back in 2021.  Sounds like an old certificate that VDH has reused for the past 3 years.  As far as I can tell, there's nothing like that attached to the xpi file for the VDH extension itself, although I know there's some sort of signature process that goes on that allows developers to register their extensions with Mozilla.  I don't know if developers have to buy certificates.  I also don't know what impact it has on anything that the CoApp's installer has an expired certificate.  It doesn't seem to be having any impact on my system.  Windows Defender hasn't made any complaints.  But then again, I haven't reinstalled the CoApp.  It would be interesting to get @paulrouget Paul's comments on this.

**Reply by Wixlogo** [reply]
I'm not referring to the certificate of the Firefox add-on/extension or the XPI file; I'm discussing the Windows executable of the program. The Windows signature certificate is unrelated to the browser add-on. The validity of the certificate does not impact the installer/program in any way; it just merely displays as a verified developer when the app/installer is run (as admin) instead of indicating the app comes from an unknown developer. There seems to be no issue otherwise. As previously stated, my intention with the post was to remind @paulrouget about the certificate renewal. Will the certificate be renewed, or not. It actually costs money to register the program to appear as a verified publisher BTW, and there was no announcement made about.

**Comment by paulrouget** [comment]
Ah damn. Working on it!

**Comment by Wixlogo** [comment]
@paulrouget updates on it, sir? 
waiting to install the program...

**Reply by GrampaWildWilly** [reply]
I would never wait to install a program because it didn't have a certificate if I had been using it for years & I downloaded it from a known good site.  The administrivia involved in applying a certificate would never deter me from installing something I find useful.  I would never deny myself the convenience of using an application just because it didn't have a certificate if I knew via other means that the application was just fine.

**Reply by Wixlogo** [reply]
Yeah, I know, but I actually run AppLocker on my system, so I have to manually allow any unsigned programs to run or install. 😅
I could whitelist the installer, but since you'll eventually provide a certificate for the program, it's better if I wait instead of creating a whitelist entry for it.
